test/e2e/util.go

```diff
@@ -229,18 +229,15 @@ func rootCredentials(kubeClient client.Client, name string) (map[string][]byte,
 // and the other CNAMEs down to the last one are not known to this replaced nameserver.
 // This may result in "no such host" error.
 func lookupCNAME(host, server string) (string, error) {
-	c := dns.Client{}
-	m := dns.Msg{}
-	if host[len(host)-1] != '.' {
-		host += "."
-	}
-	m.SetQuestion(host, dns.TypeCNAME)
-	r, _, err := c.Exchange(&m, server+":53")
+	dnsClient := dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeCNAME)
+	r, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
 	if err != nil {
 		return "", err
 	}
 	if len(r.Answer) == 0 {
-		return "", fmt.Errorf("not found")
+		return "", fmt.Errorf("no CNAME records found for host %s on %s", host, server)
```

Comment:
https://github.com/openshift/external-dns-operator/pull/92#discussion_r766123793

Reply:
Not sure whether the anchor to a resolved comment works well, let me quote it:

> I think this is a bit of an overkill as error. Here is what we are getting if this error is used:
> ```
> operator_test.go:221: Looking for DNS record in nameserver: ns-596.awsdns-10.net
> operator_test.go:227: Waiting for DNS record: myroute.1639073290.3c8d057.example-test.info, error: No results for the > host :myroute.1639073290.3c8d057.example-test.info. in nameServer : ns-596.awsdns-10.net
>```
>
>Note the redundant hostname and nameserver. They are passed to the function, so the caller always knows them and can log them if there is a need, that's what happened here.
>
>Compare the previous example to this:
>```
>operator_test.go:221: Looking for DNS record in nameserver: ns-596.awsdns-10.net
>operator_test.go:227: Waiting for DNS record: myroute.1639073290.3c8d057.example-test.info, error: not found
>```